### ISSUE TITLE
Switching off global | local validation rules (in Territory-based Validation rules)

### ISSUE THREAD

**Issue by adammertel** [DISSINET/InkVisitor#1989]
Owner should be able to switch on/off warnings

**Comment by davidzbiral** [comment]
@davidzbiral Write into https://docs.google.com/document/d/1pmsCjNLHAOA58gVrwV64Qfsp-9wO4kZoXbtyjViMuQA/edit what owner should / should not be able to switch off.

**Comment by davidzbiral** [comment]
@adammertel , I reviewed this. (Against what I said when we met, the user who has the needed rights should be able to switch off even valency warnings in Statements (mismatch of statement actants with valency frame), because they are disturbing in SumCASTEMO approach.)

Let us do this: Under each T (incl. root T, where, however, only owner can set this; under other Ts anybody who has editing rights to it set this, i.e. any admin, and editors who have this T assigned), in the "Validation rules" section, include the following checboxes (default = all checked in):
- Validate actant entity type against Action valency frames (this stands the whole section 1 of the gdoc, and allows switching it off, esp. for SumCASTEMO uses)
- Entity warnings: only in root T, not other Ts, because these are global entity warnings displayed on Detail, not contextual T-based warnings: list one by one the warning messages from the table in the gdoc (they are many, so unfold only upon click, and allow to select/unselect all), each of them as a checkbox, and allow the owner to switch them off by unchecking the box.

Note that in the table, I changed the wording of ca 3 warnings (suggestions highlighted). Please update this in InkV (I changem them in such a way that they serve better the use in this checkbox list.)

I also clarified some texts and the structure for documentation's sake. Have a look [at the gdoc](https://docs.google.com/document/d/1pmsCjNLHAOA58gVrwV64Qfsp-9wO4kZoXbtyjViMuQA/edit) and review suggestions.

One important thing that I don't known whether you noted down is:
"Only the user with owner rights can set Validation rules for root T. Validation rules for root T are global, and thus display both on Statements using entities which did not pass validation, and on Detail of those entities (at the same place where we display valency warnings for Actions)." If you haven't noted this down yet, please copy it wherever appropriate.

**Comment by davidzbiral** [comment]
@adammertel In relation to this: should we not make the Entity warning descriptions in the table more editor-readable, less technical, and display them also in this list in the app as tooltip? Happy to rewrite, if I have your agreement that it is a good idea to have the more detailed descriptions of the rules there.

**Comment by davidzbiral** [comment]
Note that my current extension, much more logical I think, even non-owner can switch some things off for his/her Territory. But not global rules, which concern Entities, of course - these only owner can switch off in the settings of root T.

**Comment by adammertel** [comment]
I have moved this issue to 1.4.1

**Comment by davidzbiral** [comment]
@adammertel Is anything more needed from me now for this issue?

**Comment by adammertel** [comment]
@davidzbiral no I think this is clear for now. Just moved to 1.4.1 so we can focus on the other things.

**Comment by adammertel** [comment]
@davidzbiral coming back to this - I think it would be a good idea to provide better human-readable non-technical descriptions to the Action valency warnings [here](https://docs.google.com/document/d/1pmsCjNLHAOA58gVrwV64Qfsp-9wO4kZoXbtyjViMuQA/edit#heading=h.qf272wpfiqhh)

**Comment by davidzbiral** [comment]
@ReimaValimaki For me to better understand: what you expect from this? That you will not get Statement-level invalid messages when doing SumCASTEMO, yes?

**Comment by davidzbiral** [comment]
@Ptrhnk I have seen this implemented in 1.4.1 - some still cannot be switched off (grayed out), but this is up to further discussion and me writing in the gdoc, the possibility is clearly there. Closing.

**Comment by Ptrhnk** [comment]
@adammertel Just to clear up things.. I renamed this to validation rule settings which are the on|off warnings. Deactivation of validation rules is implemented. Validation settings example is on this image.
<img width="653" alt="Image" src="https://github.com/user-attachments/assets/62fc5a4b-4724-42c5-a64f-aa6242ac3d41" />

**Comment by Ptrhnk** [comment]
> Note that my current extension, much more logical I think, even non-owner can switch some things off for his/her Territory. But not global rules, which concern Entities, of course - these only owner can switch off in the settings of root T.

@davidzbiral Is this still relevant?

> [@ReimaValimaki](https://github.com/ReimaValimaki) For me to better understand: what you expect from this? That you will not get Statement-level invalid messages when doing SumCASTEMO, yes?

Also if @ReimaValimaki can confirm the desired behavior so we can move on with this.

**Comment by davidzbiral** [comment]
> @davidzbiral Is this still relevant?

@Ptrhnk I don't think so - owner sets global rules, only owner can switch them off. I don't know what I meant here. So the point is: allow owner to switch off global rules instead of deleting them (probably already done, right?). Allow editor also to switch on/off territory-specific rules that s/he can set (after checking their rights for the given territory).

So rather, the point of this issue is to enable the owner to switch off even the validations which are now 100% automatic.
